mautic/mautic
# Fix for missing validator messages

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

While clicking through Mautic, I noticed a raw untranslated validator message for a form field.
As it is not straight forward to know if a translation is missing, I created a script to scan the codebase (see below)
This PR addresses all valid cases of missing / incorrect translation strings.

the script returns some false positives, but preventing those is not relevant in this case

```php
<?php

$non_existent = [];
$used_validator_messages = [];
$defined_validator_messages = [];
exec("grep -rohE \"'message'\s+=>\s+'mautic\.([A-Za-z0-9_\-]+\.[a-zA-Z0-9_\-\.]+)'\" app/bundles/* plugins/* | sort | uniq | cut -d\' -f4", $used_validator_messages);
exec("find app/bundles/* plugins/* -name 'validators.ini' -print | xargs grep '' | cut -d\: -f2 | cut -d\= -f1 | sort | uniq", $defined_validator_messages);

foreach ($used_validator_messages as $message) {
    if (!in_array($message, $defined_validator_messages)) {
        $non_existent[] = $message;
    }
}

echo var_export($non_existent, TRUE) . PHP_EOL;
```



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. validate the regex in the script
